mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -1341,6 +1358,10 @@ impl ObjectClient for S3CrtClient {
     }
 }
 
+pub trait OnTelemetry: std::fmt::Debug + Send + Sync {
```

Comment:
I've defined a trait since I find it a little more readable than raw `Fn` types, let me know if you prefer differently

Reply:
That's fair, I can swap to a type alias

Reply:
Actually this is not possible

`S3ClientConfig` requires that everything implements std::fmt::Debug, and type aliases can only implement auto-traits.